### PR TITLE
docs(cli): update docs for getStudioEnvironmentVariables

### PR DESCRIPTION
**Pull request by RitaDias**
### Description

Brush up the tsdoc comments for `getStudioEnvironmentVariables`

The reason why the params obj was moved out into its own interface and exported is due to how some limitations with TSdoc. As it was before there was a huge bulk of text on the `getStudioEnvironmentVariables` which made it hard to quickly parse through.

Now we link to an exported interface (set as public since `getStudioEnvironmentVariables` is also public).

### What to review

Make sure that the information makes sense. The text itself wasn't changed much from the previous iteration, just an example added and some words shuffled

### Notes for release
 
N/A (Not sure if we want to include this in the release)

### CODE REVIEW

File: packages/sanity/exports/cli.ts [hunk 1]
```diff
@@ -2,4 +2,5 @@ export {createCliConfig, defineCliConfig, getCliClient} from '@sanity/cli'
 
 export type {CliClientOptions, CliConfig} from '@sanity/cli'
 
+export type {studioEnvVariablesOptions} from '../src/_internal/cli/server/getStudioEnvironmentVariables'
 export {getStudioEnvironmentVariables} from '../src/_internal/cli/server/getStudioEnvironmentVariables'
```
Comment (rexxars) [inline packages/sanity/exports/cli.ts:6]: Prefer a single re-export per file:
```suggestion
export {getStudioEnvironmentVariables, type studioEnvVariablesOptions} from '../src/_internal/cli/server/getStudioEnvironmentVariables'
```

File: packages/sanity/src/_internal/cli/server/getStudioEnvironmentVariables.ts [hunk 2]
```diff
@@ -4,29 +4,46 @@ import {loadEnv} from '@sanity/cli'
...
- * in the same way the studio does. A `mode` must be specified, usually `development`
- * or `production`, which will load the corresponding `.env.development` or `.env.production`.
- * To specify where to look for the dotenv files, specify `options.envFile.envDir`.
+ * The params for the `getStudioEnvironmentVariables` function that gets Studio focused environment variables.
  *
- * Specify `options.jsonEncode` to JSON-encode the values, which is handy if you want to pass
- * this to a bundlers hardcoded defines, such as Vite's `define` or Webpack's `DefinePlugin`.
+ * @public
+ */
+export interface studioEnvVariablesOptions {
```
Comment (rexxars) [inline packages/sanity/src/_internal/cli/server/getStudioEnvironmentVariables.ts:11]: Interfaces should be PascalCase
```suggestion
export interface StudioEnvVariablesOptions {
```

File: packages/sanity/src/_internal/cli/server/getStudioEnvironmentVariables.ts [hunk 3]
```diff
@@ -4,29 +4,46 @@ import {loadEnv} from '@sanity/cli'
...
+   * this to a bundlers hardcoded defines, such as Vite's `define` or Webpack's `DefinePlugin`.
+   */
+  jsonEncode?: boolean
+}
+
+/**
+ * Get environment variables prefixed with SANITY_STUDIO_, as an object.
  *
  * @param options - Options for the environment variable loading
+ *  {@link studioEnvVariablesOptions}
```
Comment (rexxars) [inline packages/sanity/src/_internal/cli/server/getStudioEnvironmentVariables.ts:35]: Do we need this? I thought TSDoc was smart enough to know which type it was, and thus automatically link it without a specific mention of the type?

**Review by rexxars — CHANGES_REQUESTED** [review]

File: packages/sanity/src/_internal/cli/server/getStudioEnvironmentVariables.ts (hunk 3 above)
Comment (binoy14) [reply packages/sanity/src/_internal/cli/server/getStudioEnvironmentVariables.ts:35]: It does not automatically link to it. The `@link` tag would be needed for anything that we want to link it to.

Comment (binoy14) [reply packages/sanity/src/_internal/cli/server/getStudioEnvironmentVariables.ts:35]: Also updating the link when the interface is changed 
```suggestion
 *  {@link StudioEnvVariablesOptions}
```

File: packages/sanity/exports/cli.ts [hunk 4]
```diff
@@ -2,4 +2,4 @@ export {createCliConfig, defineCliConfig, getCliClient} from '@sanity/cli'
 
 export type {CliClientOptions, CliConfig} from '@sanity/cli'
 
-export {getStudioEnvironmentVariables} from '../src/_internal/cli/server/getStudioEnvironmentVariables'
+export {getStudioEnvironmentVariables, type studioEnvVariablesOptions} from '../src/_internal/cli/server/getStudioEnvironmentVariables'
```
Comment (binoy14) [inline packages/sanity/exports/cli.ts:5]: I think linter is complaining about spacing here

**Review by bjoerge — APPROVED** [review]
LGTM!

**Review by binoy14 — APPROVED** [review]
✅

**Pull request merged**